### PR TITLE
PLANET-7304 Move ENForm block to master theme

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7304](https://jira.greenpeace.org/browse/PLANET-7304)

### Testing

The block should still behave as expected, both in the backend and in the frontend. You can for example check out [this Action](https://www-dev.greenpeace.org/test-janus/petitions/oceans/) with an existing ENForm.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [janus](https://www-dev.greenpeace.org/test-janus) | [admin](https://www-dev.greenpeace.org/test-janus/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-janus/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-janus) | [composer-local.json](https://github.com/greenpeace/planet4-test-janus/blob/main/composer-local.json)

:watch: 2024.01.11 10:47:38

**Comment by sagarsdeshmukh** [comment]
Hi @mleray , Looks like some files are missing from the blocks plugin like [enforms.js ](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/admin/js/enforms.js) which is used at [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/controller/menu/class-enform-post-controller.php#L396-L403).

I deactivated the blocks plugin and tried to test the en_form from master-theme but the EN form menu option is no longer visible. Could you please recheck?

**Comment by mleray** [comment]
SonarCloud is complaining about duplicated code, but I think it's fine to leave it for now since the PR is complicated enough as it is, maybe we can refactor in the future

**Comment by sagarsdeshmukh** [comment]
image
We have 2 Api folders in the master theme. we can go with a newer i.e `src > Controllers > Api` but it's good to hear from the other team members.

**Comment by mleray** [comment]
> We have 2 Api folders in the master theme. we can go with a newer i.e `src > Controllers > Api` but it's good to hear from the other team members.

@sagarsdeshmukh I used this structure to reflect what we have in the plugin, but I wouldn't mind moving these files to the already existing Api folder.

**Comment by sagarsdeshmukh** [comment]
We also need a PR in blocks plugin to remove/disable EN form block integration.

**Comment by sagarsdeshmukh** [comment]
The `planet4-engagingnetworks` textdomain need to add in master theme.
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/class-loader.php#L537-L539

**Comment by sagarsdeshmukh** [comment]
The [p4en_vars](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/class-loader.php#L400) need to send to frontend which are used at -

https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/assets/src/blocks/ENForm/ENFormSettings.js#L180

https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/assets/src/blocks/ENForm/ENFormSettings.js#L200

File: assets/src/blocks/ENForm/ENFormFrontend.js [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+    thankyou_url,
+    setErrorMsg,
+    setActiveTplId,
+    en_page_id,
+  } = props;
+
+  const post_data = makePostData(form_data, fields);
+
+  // Send form
+  const post_url = `${window.p4bk_vars.siteUrl}/wp-json/planet4/v1/enform/${en_page_id}`;
```
Comment (sagarsdeshmukh) [inline assets/src/blocks/ENForm/ENFormFrontend.js:220]: This endpoint need to move in master theme.
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/rest/class-rest-api.php#L376-L390

Comment (sagarsdeshmukh) [reply assets/src/blocks/ENForm/ENFormFrontend.js:220]: At some point we have to move the `p4bk_vars` to master-theme as well. Could you please check if it is present in blocks plugin and master theme, does that make any issue? 🤔

Comment (mleray) [reply assets/src/blocks/ENForm/ENFormFrontend.js]: [Updated](https://github.com/greenpeace/planet4-master-theme/pull/2190/commits/b95367b2004b734d88ecfd36abb9e0398430017e)! I've moved `themeUrl` to the `p4_vars` we use in the master theme for now. For the siteUrl we can use the dataset NRO value.

**Comment by sagarsdeshmukh** [comment]
Just a note for future reference, At some point we also need to move the loco.xml's definitions for the blocks plugin and engaging networks -
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/loco.xml#L27-L50

**Comment by mleray** [comment]
> The `feature_engaging_networks` need to set from master theme, which is used at [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/assets/src/blocks/Counter/CounterBlock.js#L98). https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/class-loader.php#L390

@sagarsdeshmukh isn't this already done via [this](https://github.com/greenpeace/planet4-master-theme/pull/2190/files#diff-6b8792fb39e5bc086df0f352486bd1b87cd573422d0702519e6f77f0103c7ef3R212)?

**Comment by mleray** [comment]
> The [p4en_vars](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/class-loader.php#L400) 

@sagarsdeshmukh these [have been added to the `p4_vars`](https://github.com/greenpeace/planet4-master-theme/pull/2190/files#diff-6b8792fb39e5bc086df0f352486bd1b87cd573422d0702519e6f77f0103c7ef3R213-R214) and are used as such in the master-theme files

**Comment by mleray** [comment]
> The `planet4-engagingnetworks` textdomain need to add in master theme. 

@sagarsdeshmukh [updated](https://github.com/greenpeace/planet4-master-theme/pull/2190/files#diff-05ae9cddcaec1e845771a7db224961439f83ef5939ec67d3a48744cb34d7e58b)!

**Comment by sagarsdeshmukh** [comment]
> https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/class-loader.php#L390

Oh I overlooked this, This should work. 👍

**Comment by mleray** [comment]
> We also need a PR in blocks plugin to remove/disable EN form block integration.

@sagarsdeshmukh in the plugin we check [if the block is already registered](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/blocks/class-enform.php#L99-L101), and the ENForm backend settings are not added twice, isn't that enough? 🤔

**Comment by sagarsdeshmukh** [comment]
> > The `planet4-engagingnetworks` textdomain need to add in master theme.
> 
> @sagarsdeshmukh [updated](https://github.com/greenpeace/planet4-master-theme/pull/2190/commits/6aafb4e990415e7a509dd45f471f9cd39c9db47d)!

Here I mean,
`// Load EN translations.`
`load_plugin_textdomain( 'planet4-engagingnetworks', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );`
`load_plugin_textdomain( 'planet4-engagingnetworks-backend', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );`

**Comment by sagarsdeshmukh** [comment]
> > The `feature_engaging_networks` need to set from master theme
> 
> @sagarsdeshmukh isn't this already done via [this](https://github.com/greenpeace/planet4-master-theme/pull/2190/files#diff-6b8792fb39e5bc086df0f352486bd1b87cd573422d0702519e6f77f0103c7ef3R212)?

Oh I overlooked this, This should work. 👍

**Comment by mleray** [comment]
> Here I mean, `// Load EN translations.` `load_plugin_textdomain( 'planet4-engagingnetworks', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );` `load_plugin_textdomain( 'planet4-engagingnetworks-backend', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );`

@sagarsdeshmukh something like [this](https://github.com/greenpeace/planet4-master-theme/pull/2190/commits/bf79ba725aeee95c8f686ef9e8c3624d6dbe8f32) in the theme?

**Comment by sagarsdeshmukh** [comment]
> > Here I mean, `// Load EN translations.` `load_plugin_textdomain( 'planet4-engagingnetworks', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );` `load_plugin_textdomain( 'planet4-engagingnetworks-backend', false, P4GBKS_PLUGIN_DIRNAME . '/languages/enform/' );`
> 
> @sagarsdeshmukh something like [this](https://github.com/greenpeace/planet4-master-theme/pull/2190/commits/bf79ba725aeee95c8f686ef9e8c3624d6dbe8f32) in the theme?

kind of but the language files paths are also needed(with all language files), we can do it together with blocks plugin language files. you can ignore this change in current PR.

**Comment by sagarsdeshmukh** [comment]
When we add new EN form from backend(Engaging Networks >> Add New), and try to add a Available fields in form -
the fields are not get added and through's errors in console.
 
image

**Pull request merged**